File: design/problem_book/drinks-selling-machine/Python/base/drink_selling_machine/goods/drinks.py
from base.drink_selling_machine.goods.ingredients import Item


class PreReadyDrink(Item):
    def __init__(self, name='', amount=None):
        super().__init__(name=name)
        if amount is not None:
            self.amount: int = amount

    def __repr__(self):
        return f'Pre ready good: {self._name}'


class CookedDrink(Item):
    def __init__(self, name=''):
        super().__init__(name=name)

    def __repr__(self):
        return f'Cooked drink: {self._name}'

File: design/problem_book/drinks-selling-machine/C++/base/drink_selling_machine/goods/Drinks.cpp
#include "Drinks.h"

PreCoockedDrink::PreCoockedDrink(const std::string& name, size_t amount)
	: Item(name)
	, m_amount(amount)
{
}

std::string PreCoockedDrink::repr() const
{
	return std::string("Pre ready good: ") + m_name;
}


CoockedDrink::CoockedDrink(const std::string& name)
	: Item(name)
{
}

std::string CoockedDrink::repr()
{
	return std::string("Coocked drink: " + m_name);
}
